### PR TITLE
Bug/throw exception on enum ordinal change

### PR DESCRIPTION
**Pull request by hg-ms**
Prevent storage corruption as described in #381 and #432 by throwing an exception in case of enum ordianl changes by automatic legacy type mappings.

### CODE REVIEW

File: persistence/persistence/src/main/java/one/microstream/persistence/exceptions/PersistenceExceptionTypeConsistencyEnum.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package one.microstream.persistence.exceptions;
+
```
Comment (zdenek-jonas) [inline persistence/persistence/src/main/java/one/microstream/persistence/exceptions/PersistenceExceptionTypeConsistencyEnum.java:2]: Licence

File: persistence/persistence/src/main/java/one/microstream/persistence/types/PersistenceLegacyTypeHandlerCreator.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
 import one.microstream.collections.BulkList;
 import one.microstream.persistence.exceptions.PersistenceException;
+import one.microstream.persistence.exceptions.PersistenceExceptionTypeConsistencyEnum;
 import one.microstream.reflect.XReflect;
```
Comment (zdenek-jonas) [inline persistence/persistence/src/main/java/one/microstream/persistence/types/PersistenceLegacyTypeHandlerCreator.java:29]: fix licence please, it is mixed with imports

**Review by fh-ms — APPROVED** [review]

**Review by zdenek-jonas — APPROVED** [review]

**Comment by yqbjtu** [comment]
is this bug fix?  I have the same issue,  When I changed the enum order, I load the storage file ,get the following error:
 one.microstream.storage.embedded.types.EmbeddedStorageManager$Default: [389] Exception occured while initializing embedded storage manager
one.microstream.persistence.exceptions.PersistenceException: Inconcistency for com.yq.demo.cell.microstream.enumdemo.MyType.Type1


one.microstream.persistence.exceptions.PersistenceException: Enum constant inconsistency: in type com.yq.demo.cell.microstream.enumdemo.MyType persisted instance with ordinal 1, name "Type2" does not match JVM-created instance with ordinal 1, name "Type1"

===
the original enum:
public enum MyType {
    Type1,
    Type2,
    Type3
}
I used it in dataRoot and store the instance. 

I update it to the following:
public enum MyType {
    Type2,

    Type1,

    Type3
}

then load the storage data

**Comment by hg-ms** [comment]
Yes, since this PR an exception is thrown instead of silently accepting the ordinal change of an enum constant that can cause a corrupted storage. It’s released with 08.00.00-MS-GA.

**Pull request merged**